src/BloomExe/Edit/EditingModel.cs

```diff
@@ -671,13 +671,13 @@ public void HandleAddNewPageKeystroke(string unused)
 				if (CanDuplicatePage)
 				{
 					AddNewPageBasedOnTemplate(this._pageSelection.CurrentSelection.IdOfFirstAncestor);
-					return;
-				}
-				else
-				{
-					var idOfFirstPageInTemplateBook = CurrentBook.FindTemplateBook().GetPageByIndex(0).Id;
-					AddNewPageBasedOnTemplate(idOfFirstPageInTemplateBook);
+					if (_addNewSuccess)
```

Comment:
This doesn't warrant introducing a new member variable. Just send a return value and act on it accordingly.